### PR TITLE
Add RewardRepair

### PR DESCRIPTION
**Pull request by thanhlecongg**
Hi,

This pull request implements [RewardRepair](https://arxiv.org/pdf/2105.04123.pdf). As RewardRepair does not provide a pre-processing code, I have implemented a preprocessing code based on the preprocessing code of SelfAPR (proposed by the same authors). I have tried to contact the authors but they have not replied to me yet. Once they reply, I will create a new version with the provided preprocessing code. As recommended by @Marti2203, I name the current version as RewardRepairI to distinguish with the original version.

About runtime information:
- Running time: around 2-3 minutes for patch generation and around 30 minutes for patch validation
- In terms of memory usage, AlphaRepair took at most 1-2 Gb VRAM (for GPUs)

Please help me to review.

Thanks.

### CODE REVIEW

File: app/drivers/tools/repair/java/SelfAPR.py [hunk 1]
```diff
@@ -6,8 +6,7 @@
 
 class SelfAPR(AbstractRepairTool):
     """
-    Requirements for this tool:
-    5 GB of VRAM and 6 GB of RAM
+    Requirements for this tool: 6GB of VRAM (for GPUs)
```
Comment (Marti2203) [inline app/drivers/tools/repair/java/SelfAPR.py:9]: I don't see a reason for this change?

**Comment by Marti2203** [comment]
Hi @thanhlecongg ,
I will run it tomorrow evening or on Tuesday and get back to you by Wednesday :)

File: app/drivers/tools/repair/java/SelfAPR.py (hunk 1 above)
Comment (thanhlecongg) [reply app/drivers/tools/repair/java/SelfAPR.py:9]: It should be 6Gb of VRAM instead of 5

Comment (thanhlecongg) [reply app/drivers/tools/repair/java/SelfAPR.py:9]: The exact number is actually 5.8Gb

Comment (Marti2203) [reply app/drivers/tools/repair/java/SelfAPR.py:9]: I see, okay :D

**Comment by thanhlecongg** [comment]
Yes, please take your time. Let me know if there are any issues.

**Comment by Marti2203** [comment]
Looks good, I see that the interface is similar to the other models so I have no problems

**Pull request merged**